### PR TITLE
Set up CI, enforce gofmt

### PR DESCRIPTION
**Pull request by dbrgn**
See diff - let's hope it works :)

### CODE REVIEW

**Review by rnestler — APPROVED** [review]
LGTM

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    branches:
+      - master
+  pull_request:
+
+name: CI
+
+jobs:
+  test:
+    name: run tests
+    runs-on: ubuntu-latest
```
Comment (rnestler) [inline .github/workflows/ci.yml:12]: I usually configure a specific Ubuntu version for more reproducibility.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  fmt:
+    name: check code formatting with gofmt
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version: '^1.13.0'
+      - name: run gofmt
+        run: diff -u <(echo -n) <(gofmt -d ./)
```
Comment (rnestler) [inline .github/workflows/ci.yml:30]: Isn't it enough to just run `gofmt` with the `-d` flag?

Comment (dbrgn) [reply .github/workflows/ci.yml:30]: Nah, for some reason they seem to think that it's helpful to return exit code 0 even if there are changes...

https://stackoverflow.com/questions/42510140/check-format-for-continous-integration

File: .github/workflows/ci.yml (hunk 1 above)
Comment (dbrgn) [reply .github/workflows/ci.yml]: Updated!

File: .github/workflows/ci.yml (hunk 2 above)
Comment (rnestler) [reply .github/workflows/ci.yml:30]: I mean I'd expect it to have a `--check` flag or similar and then return an error, but it doesn't seem to have that.

Comment (dbrgn) [reply .github/workflows/ci.yml:30]: I expected such a flag as well, but it probably breaks some kind of perceived unix philosophy - why implement a feature if you can do it with convoluted shell pipelines as well? 😂

**Pull request merged**